packages/backend/src/peripherals/database/ForcedWithdrawRepository.ts

```diff
@@ -0,0 +1,368 @@
+    this.getByPositionId = this.wrapGet(this.getByPositionId)
+    this.getByStarkKey = this.wrapGet(this.getByStarkKey)
+    this.getByStateUpdateId = this.wrapGet(this.getByStateUpdateId)
+    this.getMinedNotIncluded = this.wrapGet(this.getMinedNotIncluded)
+    this.getJustSentHashes = this.wrapGet(this.getJustSentHashes)
+    this.deleteAll = this.wrapDelete(this.deleteAll)
+    this.deleteAfter = this.wrapDelete(this.deleteAfter)
+
+    /* eslint-enable @typescript-eslint/unbound-method */
+  }
+
+  async addSent(
+    record: ForcedWithdrawTransactionRecord & { timestamp: Timestamp }
+  ): Promise<Hash256> {
+    const knex = await this.knex()
+    await knex.transaction(async (trx) => {
+      await trx('forced_withdraw_transactions')
+        .insert(toRow(record))
+        .onConflict('hash')
+        .ignore()
+      await trx('forced_withdraw_statuses').insert({
+        hash: record.hash.toString(),
+        status: 'sent',
+        timestamp: BigInt(record.timestamp.toString()),
+      })
+    })
+    return record.hash
+  }
+
+  async addMined(
+    record: ForcedWithdrawTransactionRecord & {
+      timestamp: Timestamp
+      blockNumber: number
+    }
+  ): Promise<Hash256> {
+    const knex = await this.knex()
+    await knex.transaction(async (trx) => {
+      await trx('forced_withdraw_transactions')
+        .insert(toRow(record))
+        .onConflict('hash')
+        .ignore()
+      await trx('forced_withdraw_statuses').insert({
+        hash: record.hash.toString(),
+        status: 'mined',
+        timestamp: BigInt(record.timestamp.toString()),
+        block_number: record.blockNumber,
+      })
+    })
+    return record.hash
+  }
+
+  async addReverted(status: {
+    hash: Hash256
+    timestamp: Timestamp
+    blockNumber: number
+  }): Promise<Hash256> {
+    const knex = await this.knex()
+    await knex('forced_withdraw_statuses').insert({
+      hash: status.hash.toString(),
+      status: 'reverted',
+      timestamp: BigInt(status.timestamp.toString()),
+      block_number: status.blockNumber,
+    })
+    return status.hash
+  }
+
+  async addForgotten(status: {
+    hash: Hash256
+    timestamp: Timestamp
+  }): Promise<Hash256> {
+    const knex = await this.knex()
+    await knex('forced_withdraw_statuses').insert({
+      hash: status.hash.toString(),
+      status: 'forgotten',
+      timestamp: BigInt(status.timestamp.toString()),
+    })
+    return status.hash
+  }
+
+  async addIncluded(status: {
+    hash: Hash256
+    timestamp: Timestamp
+    blockNumber: number
+    stateUpdateId: number
+  }): Promise<Hash256> {
+    const knex = await this.knex()
+    await knex('forced_withdraw_statuses').insert({
+      hash: status.hash.toString(),
+      status: 'included',
+      timestamp: BigInt(status.timestamp.toString()),
+      block_number: status.blockNumber,
+      state_update_id: status.stateUpdateId,
+    })
+    return status.hash
+  }
```

Comment:
What's the difference between included and mined transactions?

Reply:
Mined means that the transaction appeared on the blockchain. Included means that a state update referenced that particular forced transaction. That usually happens several hours later. It's only something relevant to forced transactions in StarkEx

Reply:
Ok, makes sense now, thanks